python/pyarrow/dataset.py

```diff
@@ -81,9 +81,13 @@ def partitioning(field_names=None, flavor=None):
 
     Parameters
     ----------
-    field_names : pyarrow.Schema or list of str
-        The schema that describes the partitions present in the file path. If
-        a list of strings (field names) is passed, the schema's types are
+    schema : pyarrow.Schema, default None
+        The schema that describes the partitions present in the file path.
+        If not specified, and `field_names` and/or `flavor` are specified,
+        the schema will be inferred from the file path (and a
+        PartitioningFactory is returned).
+    field_names :  list of str, default None
+        A list of strings (field names). If specified, the schema's types are
```

Comment:
@bkietz @kszucs following up on https://github.com/apache/arrow/pull/6022#discussion_r367016454,  I went with splitting the keyword into two separate ones. 
But I can also rename it to `schema_or_field_names` and keep it as a single one (it's long, but you can use it positionally), if that has stronger preference. 

I think having separate ones is a bit cleaner, and easier to explain.

Reply:
And it is easier to merge than separate later.